2-0-data-structures-and-algorithms/src/test/java/com/bobocode/linked_queue/LinkedQueueTest.java

```diff
@@ -1,103 +1,372 @@
+        integerQueue.add(1);
+        integerQueue.add(228);
+        int size = getInternalSize();
+        boolean isEmpty = isEmptyQueue();
+        Integer firstElement = (Integer) pollElementFromQueue();
+        Integer secondElement = (Integer) pollElementFromQueue();
 
-        assertEquals(3, integerQueue.size());
-        assertEquals(324, integerQueue.poll().intValue());
+        assertThat(size).isEqualTo(2);
+        assertThat(isEmpty).isEqualTo(false);
+        assertThat(firstElement).isEqualTo(1);
+        assertThat(secondElement).isEqualTo(228);
     }
 
     @Test
     @Order(5)
-    void pollElementFromEmptyQueue() {
-        assertNull(integerQueue.poll());
+    void addFillsQueueWhenItIsNotEmpty() {
+        addIntElementToQueue(12);
+        integerQueue.add(111);
+        int size = getInternalSize();
+        boolean isEmpty = isEmptyQueue();
+        Integer firstElement = (Integer) pollElementFromQueue();
+        Integer secondElement = (Integer) pollElementFromQueue();
+
+        assertThat(size).isEqualTo(2);
+        assertThat(isEmpty).isEqualTo(false);
+        assertThat(firstElement).isEqualTo(12);
+        assertThat(secondElement).isEqualTo(111);
     }
 
     @Test
     @Order(6)
-    void pollElement() {
-        integerQueue.add(33);
-        integerQueue.add(123);
-        integerQueue.add(222);
-        integerQueue.add(444);
-
-        integerQueue.poll(); // should poll 33
+    void addIncreasesQueueSize() {
+        integerQueue.add(1);
+        integerQueue.add(2);
+        int size = this.integerQueue.size();
 
-        assertEquals(123, integerQueue.poll().intValue());
-        assertEquals(2, integerQueue.size());
+        assertThat(size).isEqualTo(2);
     }
 
     @Test
     @Order(7)
-    void pollLastElement() {
-        integerQueue.add(8);
-        integerQueue.add(123);
-        integerQueue.add(99);
-        integerQueue.add(46);
-
-        integerQueue.poll(); // should poll 8
-        integerQueue.poll(); // should poll 123
-        integerQueue.poll(); // should poll 99
+    void isEmptyReturnsFalseWhenElementIsAdded() {
+        integerQueue.add(1);
+        boolean isEmpty = integerQueue.isEmpty();
 
-        assertEquals(46, integerQueue.poll().intValue());
-        assertEquals(0, integerQueue.size());
+        assertThat(isEmpty).isEqualTo(false);
     }
 
     @Test
     @Order(8)
-    void isEmptyOnEmptyQueue() {
-        assertTrue(integerQueue.isEmpty());
+    void pollReturnsNullWhenQueueIsEmpty() {
+        Integer firstElement = this.integerQueue.poll();
+
+        assertThat(firstElement).isEqualTo(null);
     }
 
     @Test
     @Order(9)
-    void isEmpty() {
-        integerQueue.add(3);
-        integerQueue.add(9);
+    void poll() {
+        addIntElementToQueue(11);
+        addIntElementToQueue(111);
+        Integer firstElement = this.integerQueue.poll();
+        Integer secondElement = this.integerQueue.poll();
+        int size = getInternalSize();
```

Comment:
Now, when we have test `pollDecreasesQueueSize()` we can delete checking of size in this test and name the test `pollDeletesElementFromHead()`. This will get users more clear understanding of theirs mistakes.